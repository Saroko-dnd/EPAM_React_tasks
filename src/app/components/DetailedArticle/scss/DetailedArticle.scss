@import "./variables";
@import "./animations";

.detailed-news-article {
  height: 60vh;
  min-height: 35rem;
  background-size: cover;
  background-position: 50% 0;

  > * {
    background-color: $text-background-color;
    &:not(.url) {
      font-weight: bold;
    }
  }

  .title {
    @include fadeinAnimation(0s);
    font-size: 2.5rem;
    color: $title-font-color;
  }

  .photo {
    display: none;
    max-height: 500px;
    @include fadeinAnimation(1s);
    animation-fill-mode: forwards;
  }

  .description {
    @include fadeinAnimation(2s);
    font-size: 2rem;
    color: $description-font-color;
  }

  .url {
    @include fadeinAnimation(3s);
    color: $link-color;
    font-size: 3rem;
  }

  footer {
    @include fadeinAnimation(4s);
    color: $footer-font-color;
  }
}
