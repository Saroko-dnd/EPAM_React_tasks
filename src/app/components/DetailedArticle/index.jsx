import React from 'react';

import constants from '../../constants';

import './scss/DetailedArticle.scss';

const DetailedArticle = ({ article }) => (
    <article className="detailed-news-article d-flex align-items-center flex-column">
        <h2 className="title">{article.title}</h2>
        <img className="photo" src={article.urlToImage} alt=":(" />
        <p className="description">{article.description}</p>
        <a className="url" href={article.url}>
            read source
        </a>
        <footer className="d-flex align-items-center flex-column">
            <p>{new Date(Date.parse(article.publishedAt)).toDateString()}</p>
            <span>{article.source.name}</span>
            <p>{article.author}</p>
        </footer>
    </article>
);

DetailedArticle.propTypes = {
    article: constants.customPropTypes.article,
};

DetailedArticle.defaultProps = {
    article: {
        source: { name: '' },
        title: '',
        author: '',
        description: '',
        urlToImage: '',
        publishedAt: '',
        url: '',
    },
};

export default DetailedArticle;
