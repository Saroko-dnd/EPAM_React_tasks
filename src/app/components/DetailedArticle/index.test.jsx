import DetailedArticle from '.';

describe('render a DetailedArticle', () => {
    const detailedArticle = mount(<DetailedArticle />);

    it('should render DetailedArticle correctly', () => {
        expect(detailedArticle).toMatchSnapshot();
    });

    it('should have correct default props', () => {
        expect(detailedArticle.prop('article')).toContainEntries([
            ['source', { name: '' }],
            ['title', ''],
            ['author', ''],
            ['description', ''],
            ['urlToImage', ''],
            ['publishedAt', ''],
            ['url', ''],
        ]);
    });
});
