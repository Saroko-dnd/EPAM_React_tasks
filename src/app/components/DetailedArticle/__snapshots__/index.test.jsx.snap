// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a DetailedArticle should render DetailedArticle correctly 1`] = `
<DetailedArticle
  article={
    Object {
      "author": "",
      "description": "",
      "publishedAt": "",
      "source": Object {
        "name": "",
      },
      "title": "",
      "url": "",
      "urlToImage": "",
    }
  }
>
  <article
    className="detailed-news-article d-flex align-items-center justify-content-center flex-column"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  >
    <div
      className="news-content"
    >
      <h2
        className="title"
      />
      <div
        className="horizontal-line d-none d-sm-block"
      />
      <p
        className="description d-none d-sm-block"
      />
    </div>
    <div
      className="place-holder"
    />
    <a
      className="url"
      href=""
    >
      read source
    </a>
    <footer
      className="d-flex align-items-center"
    >
      <p>
        Invalid Date
      </p>
      <span />
      <p
        className="d-none d-sm-block"
      />
    </footer>
  </article>
</DetailedArticle>
`;
